haystack/components/generators/chat/openai.py

```
@@ -416,14 +454,35 @@ def _prepare_api_call(  # noqa: PLR0913
 
         is_streaming = streaming_callback is not None
         num_responses = generation_kwargs.pop("n", 1)
+
+        if response_format and not is_streaming:
+            # for structured outputs without streaming, we use openai's parse endpoint
+            # Note: `stream` cannot be passed to chat.completions.parse
+            # we pass a key `openai_endpoint` as a hint to the run method to use the parse endpoint
+            # this key will be removed before the API call is made
+            return {
+                "model": self.model,
+                "messages": openai_formatted_messages,
+                "n": num_responses,
+                "response_format": response_format,
+                "openai_endpoint": "parse",
+                **openai_tools,
+                **generation_kwargs,
+            }
+
         if is_streaming and num_responses > 1:
             raise ValueError("Cannot stream multiple responses, please set n=1.")
```

I would move this simple check at the beginning of the method, to fail fast